Update wallet configuration defaults and in console wallet

philipr-za opened the pull request:
## Description

This PR syncs the default config values in the Wallet Config structs and the app config bootstrap defaults. The PR also updates the tari_console_wallet to actually use the values from the bootstrap config which was previously ignored.

## How Has This Been Tested?
Existing tests

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
* [x] Bug fix (non-breaking change which fixes an issue)
* [ ] New feature (non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to change)
* [ ] Feature refactor (No new feature or functional changes, but performance or technical debt improvements)
* [ ] New Tests
* [ ] Documentation

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
* [x] I'm merging against the `development` branch.
* [x] I ran `cargo-fmt --all` before pushing.
* [x] I ran `cargo test` successfully before submitting my PR.
* [x] I have squashed my commits into a single commit.
* [ ] My change requires a change to the documentation.
* [ ] I have updated the documentation accordingly.
* [ ] I have added tests to cover my changes.

hansieodendaal commented on base_layer/wallet/src/wallet.rs:
```diff
@@ -85,29 +85,33 @@ pub struct WalletConfig {
-            // This is the default rate limit fot the pubsub_connector for the mobile wallet
-            rate_limit: 5,
+            output_manager_service_config,
+            buffer_size: buffer_size.unwrap_or_else(|| 1500),
+            rate_limit: rate_limit.unwrap_or_else(|| 1000),
```
I think we should be careful with these; it will impact the memory footprint in the mobile wallet. The rate limit of `1000` proposed here is not balanced with the buffer size; better to throttle it a bit and not have messages dropped under stress conditions. I would propose `50`, then the `1500` buffer will fill up in 30 s if nothing is processed.

hansieodendaal commented on common/src/configuration/utils.rs:
```diff
@@ -80,13 +80,13 @@ pub fn default_config(bootstrap: &ConfigBootstrap) -> Config {
     .unwrap();
-    cfg.set_default("wallet.base_node_query_timeout", 900).unwrap();
-    cfg.set_default("wallet.transaction_broadcast_monitoring_timeout", 600)
+    cfg.set_default("wallet.base_node_query_timeout", 60).unwrap();
+    cfg.set_default("wallet.transaction_broadcast_monitoring_timeout", 60)
```
These values should be in line with what is documented in the 2x `*.toml` files, or vice versa:
```
# This is the timeout period that will be used to monitor TXO queries to the base node (default = 30). Larger values
# are needed for wallets with many (>1000) TXOs to be validated.
base_node_query_timeout = 120
# This is the timeout period that will be used for base node broadcast monitoring tasks (default = 30)
#transaction_broadcast_monitoring_timeout = 30
# This is the timeout period that will be used for chain monitoring tasks (default = 30)
#transaction_chain_monitoring_timeout = 30
# This is the timeout period that will be used for sending transactions directly (default = 20)
#transaction_direct_send_timeout = 20
# This is the timeout period that will be used for sending transactions via broadcast mode (default = 30)
#transaction_broadcast_send_timeout = 30
```
Please align the `*.toml` files with the config defaults.

hansieodendaal commented on common/src/configuration/utils.rs:
```diff
@@ -80,13 +80,13 @@ pub fn default_config(bootstrap: &ConfigBootstrap) -> Config {
+    cfg.set_default("wallet.transaction_broadcast_monitoring_timeout", 60)
         .unwrap();
-    cfg.set_default("wallet.transaction_chain_monitoring_timeout", 15)
+    cfg.set_default("wallet.transaction_chain_monitoring_timeout", 60)
```
See above

hansieodendaal commented on common/src/configuration/utils.rs:
```diff
@@ -80,13 +80,13 @@ pub fn default_config(bootstrap: &ConfigBootstrap) -> Config {
         .unwrap();
-    cfg.set_default("wallet.transaction_direct_send_timeout", 600).unwrap();
-    cfg.set_default("wallet.transaction_broadcast_send_timeout", 600)
+    cfg.set_default("wallet.transaction_direct_send_timeout", 20).unwrap();
+    cfg.set_default("wallet.transaction_broadcast_send_timeout", 60)
```
See above

hansieodendaal reviewed (commented):
Looking good, definitely fixing a previous oversight. Just some specific comments belo.

philipr-za committed:
Update wallet configuration defaults and in console wallet

This PR syncs the default config values in the Wallet Config structs and the app config bootstrap defaults. The PR also updates the tari_console_wallet to actually use the values from the bootstrap config which was previously ignored.

hansieodendaal reviewed (approved):
LGTM - untested as it stands, but will put it through its paces in the next stress test.
